File: backend/src/routes/team_routes.rs
use std::sync::Arc;

use axum::{routing::{get, post}, Router};
use crate::{handlers::team_handler, models::auth_model::AppState};

pub fn routes() -> Router<Arc<AppState>>{
    Router::new()
        .route("/", get(team_handler::fetch_teams)
            .post(team_handler::add_team))
        .route("/{id}",get(team_handler::fetch_team)
            .put(team_handler::alter_team)
            .delete(team_handler::remove_team))
        .route("/{id}/{user_id}", post(team_handler::add_to_team)
            .delete(team_handler::remove_from_team)
            .put(team_handler::alter_team_lead))
}

File: frontend/src/api/teamRoutes.js

import { apiClient } from './apiClient';

export const user_routes = {
  //Teamdata e userdata sao jsons,
  fetchTeams: () => {
    return apiClient.request("/api/team","GET"); 
  },
  //this makes a team, so tou a copiar o nome do codigo do backend
  addTeam: (teamdata) => {
    return apiClient.request(`/api/team`,"POST",teamdata);
  },

  fetchTeam: (id) => {
    return apiClient.request(`/api/team/${id}`,"GET");
  },
  
  alterTeam: (id,teamdata) => {
    return apiClient.request(`/api/team/${id}`,"PUT",teamdata);
  },

  removeTeam: (id) => {
    return apiClient.request(`/api/team/${id}`,"DELETE");
  },

  addToTeam: (id,userid,userdata) => {
    return apiClient.request(`/api/team/${id}/${userid}`,"POST",userdata);
  },
  
  removeFromTeam: (id,userid) => {
    return apiClient.request(`/api/team/${id}/${userid}`,"DELETE");
  },

  alterTeamLead: (id,userid,userdata) => {
    return apiClient.request(`/api/team/${id}/${userid}`,"PUT",userdata);
  },
};